### PR TITLE
Feature/invoice 2 use billing preprocessor

### PR DESCRIPTION
**Pull request by hensankom**
Implementera faktureringslogik: använd BillingPreProcessor, hämta information från ärende, lista fakturor på översiktssidan osv.

Denna PR ska till develop efter att [https://github.com/Sundsvallskommun/web-app-draken-public/pull/113](https://github.com/Sundsvallskommun/web-app-draken-public/pull/113) har mergats in, då den PR:en inför rollhantering vilken faktureringen är beroende av.

## Types of changes

- [ ] Bug fix
- [x] New feature
- [ ] Removed feature
- [ ] Code style update (formatting etc.)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Content/Data

## Does this PR introduce a breaking change?
- [ ] Yes (I have stepped the version number accordingly)
- [x] No
      
## Checklist:

- [x] My code follows the code style of this project.
- [ ] I have updated the documentation accordingly (if applicable).
- [ ] I have added/updated tests to cover my changes (if applicable).

### CODE REVIEW

File: frontend/src/supportmanagement/components/attestation-tab/components/attestation-filtering/components/attestation-filter-status.component.tsx [hunk 1]
```diff
@@ -11,23 +13,16 @@ export const AttestationStatusValues = {
...
-  },
-  {
-    id: 1,
-    status: 'NONE',
-  },
-];
-
 export const AttestationFilterStatusComponent: React.FC = () => {
   const { register } = useFormContext();
+  const ss = useMemo(() => {
```
Comment (danneSvall) [inline frontend/src/supportmanagement/components/attestation-tab/components/attestation-filtering/components/attestation-filter-status.component.tsx:18]: Kanske byta till ett mer beskrivande namn vad den är till för?

File: frontend/src/supportmanagement/components/attestation-tab/components/attestations-table.component.tsx [hunk 2]
```diff
@@ -280,6 +251,7 @@ export const AttestationsTable: React.FC<{
                 pages={totalPages}
                 activePage={page + 1}
                 changePage={(page) => {
+                  console.log('Setting page to', page);
```
Comment (danneSvall) [inline frontend/src/supportmanagement/components/attestation-tab/components/attestations-table.component.tsx:254]: ta bort loggning hos klienten

**Review by danneSvall — CHANGES_REQUESTED** [review]

File: backend/src/controllers/billing.controller.ts [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    //     const guidUrl = `citizen/2.0/${query}/guid`;
+    //     guidRes = await this.apiService.get<string>({ url: guidUrl }, req.user).catch(e => null);
+    //   }
+    //   let queryFilter = `(`;
+    //   queryFilter += `description~'*${query}*'`;
+    //   queryFilter += ')';
+    //   filterList.push(queryFilter);
+    // }
+    if (invoiceType) {
+      const ss = invoiceType.split(',').map(s => `invoice.description:'${s}'`);
```
Comment (timlindsk) [inline backend/src/controllers/billing.controller.ts:57]: icke-beskrivande variabelnamn

File: frontend/src/common/components/main-errands-sidebar/main-errands-sidebar.component.tsx [hunk 4]
```diff
@@ -100,15 +100,17 @@ export const MainErrandsSidebar: React.FC<{
               <Button
                 onClick={() => setShowAttestationTable(true)}
                 leftIcon={<LucideIcon name="square-pen" />}
-                className="w-full text-right justify-between"
+                // className="w-full text-right justify-between"
```
Comment (timlindsk) [inline frontend/src/common/components/main-errands-sidebar/main-errands-sidebar.component.tsx:103]: remove unused

File: frontend/src/supportmanagement/components/attestation-tab/attestation-tab.component.tsx [hunk 5]
```diff
@@ -59,6 +64,19 @@ export const AttestationTab = () => {
   const router = useRouter();
   const { user, setUser } = useAppContext();
 
+  // useAttestationFilter(
```
Comment (timlindsk) [inline frontend/src/supportmanagement/components/attestation-tab/attestation-tab.component.tsx:67]: why is this code commented out?

File: frontend/src/supportmanagement/components/attestation-tab/attestation-tab.component.tsx [hunk 6]
```diff
@@ -105,37 +123,35 @@ export const AttestationTab = () => {
...
 
   useEffect(() => {
-    if (errands) {
+    if (billingRecords) {
       setSupportErrand(undefined);
-      setTableValue('page', errands.page);
-      setTableValue('size', errands.size);
-      setTableValue('totalPages', errands.totalPages);
-      setTableValue('totalElements', errands.totalElements);
+      // setTableValue('page', billingRecords.page);
```
Comment (timlindsk) [inline frontend/src/supportmanagement/components/attestation-tab/attestation-tab.component.tsx:133]: why is this code commented out?

File: frontend/src/supportmanagement/components/attestation-tab/attestation-tab.component.tsx [hunk 7]
```diff
@@ -105,37 +123,35 @@ export const AttestationTab = () => {
...
-    // });
     getSupportAdmins().then(setSupportAdmins);
-    //eslint-disable-next-line
   }, []);
 
   useDebounceEffect(
     () => {
       const fObj = {};
-      const extraFilterObj = {};
+      // const extraFilterObj = {};
```
Comment (timlindsk) [inline frontend/src/supportmanagement/components/attestation-tab/attestation-tab.component.tsx:149]: why is this and row 168 commented out?

File: frontend/src/supportmanagement/components/attestation-tab/components/attestation-filtering/attestations-filtering.component.tsx [hunk 8]
```diff
@@ -77,7 +77,7 @@ export const AttestationsFilteringComponent: React.FC<{
           </div>
         </div>
 
-        <div className="flex flex-col md:flex-row justify-between gap-16">
+        {/* <div className="flex flex-col md:flex-row justify-between gap-16">
```
Comment (timlindsk) [inline frontend/src/supportmanagement/components/attestation-tab/components/attestation-filtering/attestations-filtering.component.tsx:80]: why is this commented out?

File: backend/src/controllers/billing.controller.ts (hunk 3 above)
Comment (hensankom) [reply backend/src/controllers/billing.controller.ts]: Åtgärdat tidigare

File: frontend/src/common/components/main-errands-sidebar/main-errands-sidebar.component.tsx (hunk 4 above)
Comment (hensankom) [reply frontend/src/common/components/main-errands-sidebar/main-errands-sidebar.component.tsx:103]: Done!

File: frontend/src/supportmanagement/components/attestation-tab/attestation-tab.component.tsx (hunk 5 above)
Comment (hensankom) [reply frontend/src/supportmanagement/components/attestation-tab/attestation-tab.component.tsx:67]: Done

File: frontend/src/supportmanagement/components/attestation-tab/attestation-tab.component.tsx (hunk 6 above)
Comment (hensankom) [reply frontend/src/supportmanagement/components/attestation-tab/attestation-tab.component.tsx:133]: Done

File: frontend/src/supportmanagement/components/attestation-tab/components/attestation-filtering/attestations-filtering.component.tsx (hunk 8 above)
Comment (hensankom) [reply frontend/src/supportmanagement/components/attestation-tab/components/attestation-filtering/attestations-filtering.component.tsx:80]: Done

File: frontend/src/supportmanagement/components/attestation-tab/attestation-tab.component.tsx (hunk 7 above)
Comment (hensankom) [reply frontend/src/supportmanagement/components/attestation-tab/attestation-tab.component.tsx:149]: Done

File: frontend/src/supportmanagement/components/attestation-tab/components/attestation-filtering/components/attestation-filter-invoice-type.component.tsx [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+export const AttestationInvoiceTypeValues = {
+  invoiceType: [],
+};
+
+export const AttestationFilterInvoiceTypeComponent: React.FC = () => {
+  const { register, getValues } = useFormContext<AttestationInvoiceTypeFilter>();
+
+  useEffect(() => {
```
Comment (timlindsk) [inline frontend/src/supportmanagement/components/attestation-tab/components/attestation-filtering/components/attestation-filter-invoice-type.component.tsx:19]: unecessary useEffect with only console log

**Review by timlindsk — APPROVED** [review]

**Review by danneSvall — APPROVED** [review]

**Pull request merged**